webrtc.html

```diff
@@ -9287,7 +9287,15 @@ <h4 id="sctp-transport-connected">Connected procedure</h4>
                   <p>Let <var>channel</var> be the <code><a>RTCDataChannel</a></code> object.</p>
                 </li>
                 <li>
-                  <p>If the <var>channel</var>'s <a>[[\DataChannelId]]</a> slot is greater or
+                  <p>If <var>channel</var>'s <a>[[\DataChannelId]]</a> slot is
+                    <code>null</code>, initialize <a>[[\DataChannelId]]</a>
+                    to a value generated by the
+                    user agent, according to [[!RTCWEB-DATA-PROTOCOL]].
+                    If no available ID could be generated,
+                    <a>throw</a> an <code>OperationError</code> exception.
+                  </p>
+                <li>
+                  <p>If <var>channel</var>'s <a>[[\DataChannelId]]</a> slot is greater or
                   equal to <var>transport</var>'s <a>[[\MaxChannels]]</a> slot,
                   <a data-lt="data-channel-failure"> close the <var>channel</var> due to a failure
                   </a>. Otherwise, <a data-lt="announce the rtcdatachannel as open">announce the
```

Comment:
Not new in this PR: We are inside of steps invoking "announce the rtcdatachannel as open" and "close the channel due to failure". The referenced algorithm queues a task. We should refactor this so that we can invoke the algorithms without queuing a task; the queuing of a task is in response to something happening in parallel.

Reply:
Not addressed in this PR.